src/main/java/org/javarosa/core/model/CoreModelModule.java

```diff
@@ -29,6 +29,7 @@
             "org.javarosa.core.model.QuestionDef",
             "org.javarosa.core.model.RangeQuestion",
             "org.javarosa.core.model.GroupDef",
+            "org.javarosa.core.model.instance.TreeReference",
```

Comment:
`TreeReferences` are already serialized in e.g. `XPathReference` or `Triggerable` but the type is passed inline. I think this is necessary to have a collection of `TreeReference`s that is serialized.